src/com/company/Parse.java

```diff
@@ -0,0 +1,120 @@
+package com.company;
+
+import org.eclipse.cdt.core.dom.ast.ExpansionOverlapsBoundaryException;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+
+    private IASTTranslationUnit translationUnit; //current translation unit node
+    public HashMap<String, Contents> repo;
+
+    public Parse(String code) throws Exception {
+        repo = new HashMap<>();
+
+        translationUnit = getIASTTranslationUnit(code.toCharArray());
+        IASTNode node = translationUnit;
+
+        saveNodes(node);
+    }
+
+    /**Accesses all the nodes of the tree*/
+    private void saveNodes(IASTNode node) throws Exception {
+        IASTNode[] children = node.getChildren();
+        for (IASTNode child :children) {
+            if (child instanceof CPPASTSimpleDeclaration) {
+
+                /**Setup*/
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+                CPPASTNamedTypeSpecifier specifier = (CPPASTNamedTypeSpecifier) declaration.getDeclSpecifier();
+                CPPASTDeclarator declarator = (CPPASTDeclarator) declaration.getDeclarators()[0];
+                CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+
+                if (init == null) {
+                    continue;
+                }
+
+                if (declarator == null) {
+                    continue;
+                }
+
+                if (specifier == null) {
+                    continue;
+                }
+
+                if (declaration == null) {
+                    continue;
+                }
+
+                CPPASTLiteralExpression value = (CPPASTLiteralExpression) init.getInitializerClause(); //get value
+                CPPASTName name = (CPPASTName) declarator.getName(); //get name
+                IToken typeQualifier = specifier.getSyntax(); //get typeQualifer type (iToken)
+                CPPASTName type = (CPPASTName) specifier.getName(); //get type
+
+                Contents contents = new Contents(name, value, typeQualifier, type);
+                repo.put(name.getRawSignature(), contents);
+            }
+        }
+    }
+
+    public Contents get(String key) throws IllegalAccessError{
+        if (repo.get(key) == null) {
+            throw new IllegalAccessError("The name does not exist!");
+        }
```

Comment:
It might be cleaner to use an assertion here, though that needs some project-wide configs. Let's get #1 merged first, then see if we can work an assertion in here.

Actually, it may be better for this to return `null` - especially since that behavior would be consistent with how `HashMap` handles things, which will be what Java developers will be used to.